pkg/detectors/azuresearchadminkey/azuresearchadminkey.go

```diff
@@ -23,79 +22,95 @@ var _ detectors.Detector = (*Scanner)(nil)
 
 var (
 	defaultClient = common.SaneHttpClient()
-	// Make sure that your group is surrounded in boundary characters such as below to reduce false positives.
-	keyPat     = regexp.MustCompile(detectors.PrefixRegex([]string{"azure"}) + `\b([0-9a-zA-Z]{52})\b`)
-	servicePat = regexp.MustCompile(detectors.PrefixRegex([]string{"azure"}) + `\b([0-9a-zA-Z]{7,40})\b`)
+
+	servicePat = regexp.MustCompile(`\b([a-z0-9][a-z0-9-]{5,58}[a-z0-9])\.search\.windows\.net`)
+	keyPat     = regexp.MustCompile(detectors.PrefixRegex([]string{"azure", "windows.net"}) + `\b([a-zA-Z0-9]{52})\b`)
 )
 
 // Keywords are used for efficiently pre-filtering chunks.
 // Use identifiers in the secret preferably, or the provider name.
 func (s Scanner) Keywords() []string {
-	return []string{"azure"}
+	return []string{"search.windows.net"}
+}
+
+func (s Scanner) Type() detectorspb.DetectorType {
+	return detectorspb.DetectorType_AzureSearchAdminKey
+}
+
+func (s Scanner) Description() string {
+	return "Azure Search is a search-as-a-service solution that allows developers to incorporate search capabilities into their applications. Azure Search Admin Keys can be used to manage and query search services."
 }
 
 // FromData will find and optionally verify AzureSearchAdminKey secrets in a given set of bytes.
 func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (results []detectors.Result, err error) {
 	dataStr := string(data)
 
-	matches := keyPat.FindAllStringSubmatch(dataStr, -1)
-	serviceMatches := servicePat.FindAllStringSubmatch(dataStr, -1)
-
-	for _, match := range matches {
-		if len(match) != 2 {
+	// Deduplicate results.
+	serviceMatches := make(map[string]struct{})
+	for _, matches := range servicePat.FindAllStringSubmatch(dataStr, -1) {
+		serviceMatches[matches[1]] = struct{}{}
+	}
+	keyMatches := make(map[string]struct{})
+	for _, matches := range keyPat.FindAllStringSubmatch(dataStr, -1) {
+		k := matches[1]
+		if detectors.StringShannonEntropy(k) < 4 {
 			continue
 		}
-		resMatch := strings.TrimSpace(match[1])
-		for _, serviceMatch := range serviceMatches {
-			if len(serviceMatch) != 2 {
-				continue
-			}
-			resServiceMatch := strings.TrimSpace(serviceMatch[1])
+		keyMatches[k] = struct{}{}
+	}
 
-			s1 := detectors.Result{
+	for key := range keyMatches {
+		for service := range serviceMatches {
+			r := detectors.Result{
 				DetectorType: detectorspb.DetectorType_AzureSearchAdminKey,
-				Raw:          []byte(resMatch),
-				RawV2:        []byte(resMatch + resServiceMatch),
+				Raw:          []byte(key),
+				RawV2:        []byte(`{"service":"` + service + `","key":"` + key + `"}`),
```

Comment:
Changing the `RawV2` value will break some enterprise customers, as we currently have secrets verified using the old `key+service` format. This change would orphan previously verified credentials and create duplicates. Is there a reason we can't continue supporting the old format?

Reply:
> Is there a reason we can't continue supporting the old format?

Obviously you get the final say here, but #2128 has been a PITA for me and others using OSS.

TL;DR Most of the existing `RawV2` values are suboptional because they smash together text (sometimes without a delimiter) and are devoid of context/keywords. This makes it impossible to programmatically re-construct and re-verify results.

Reply:
My guess is that when `RawV2` was proposed, we didn't account for its visibility in OSS reporting. Wonder if we can find a solution that preserves readability for OSS while avoiding duplicates on the enterprise side?

This likely requires a broader discussion, and I think @zricethezav  will have the most input, but I agree that combining multiple credentials makes them difficult to interpret and largely ineffective

Reply:
> My guess is that when RawV2 was proposed, we didn't account for its visibility in OSS reporting. Wonder if we can find a solution that preserves readability for OSS while avoiding duplicates on the enterprise side?

I can't speak to the impact on enterprise. I can revert the change to `RawV2`, though I'd prefer to improve it (eventually).